opensearch/src/test/java/zipkin2/storage/elasticsearch/OpensearchContainer.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright The OpenZipkin Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package zipkin2.storage.elasticsearch;
+
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.ClientOptions;
+import com.linecorp.armeria.client.ClientOptionsBuilder;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.client.WebClientBuilder;
+import com.linecorp.armeria.client.logging.ContentPreviewingClient;
+import com.linecorp.armeria.client.logging.LoggingClient;
+import com.linecorp.armeria.client.logging.LoggingClientBuilder;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.logging.LogLevel;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import zipkin2.Span;
+import zipkin2.dependencies.opensearch.OpensearchDependenciesJob;
+import zipkin2.elasticsearch.ElasticsearchStorage;
+
+import static org.testcontainers.utility.DockerImageName.parse;
+import static zipkin2.storage.ITDependencies.aggregateLinks;
+
+class OpensearchContainer extends GenericContainer<OpensearchContainer> {
+  static final Logger LOGGER = LoggerFactory.getLogger(OpensearchContainer.class);
+
+  OpensearchContainer(int majorVersion) {
+    super(parse("ghcr.io/openzipkin/zipkin-opensearch" + majorVersion + ":3.3.0"));
+    addExposedPort(9200);
+    waitStrategy = Wait.forHealthcheck();
+    withLogConsumer(new Slf4jLogConsumer(LOGGER));
+  }
+
+  @Override public void start() {
+    super.start();
+    LOGGER.info("Using baseUrl http://" + hostPort());
+  }
+
+  ElasticsearchStorage.Builder newStorageBuilder() {
+
+    WebClientBuilder builder = WebClient.builder("http://" + hostPort())
+      // Elasticsearch 7 never returns a response when receiving an HTTP/2 preface instead of the
```

Comment:
if this is valid for OS as well, adjust comment or remove special casing if it isn't?

Reply:
I will double check but OS is at the core fork of ES 7 , update your shortly, thanks!

Reply:
UPD: seems not to be relevant anymore (at lease since ES 7.10.2 and OS 2.x)

Reply:
awesome